### PR TITLE
Add error handling and logging in R Server

### PR DESCRIPTION
**Pull request by HaozhouWang**
1. Support different mode for log output
   `1` is standalone mode ->  stdout
   `2` is container mode -> send back to GPDB
   `3` is container debug mode -> write to file

2. Support error handling and exception in R server

### CODE REVIEW

File: include/rserver.hh [hunk 1]
```diff
@@ -39,25 +43,31 @@ class RServerRPC final : public PLContainer::Service {
...
+    RServerLog *rLog;
 };
 
 class RServer {
    public:
-    explicit RServer(bool standAloneMode) { this->standAloneMode = standAloneMode; }
+    RServer(bool standAloneMode, RServerLog *rLog) {
+        this->standAloneMode = standAloneMode;
+        this->rLog = rLog;
+        server = new RServerRPC(rLog);
```
Comment (wengyanqing) [inline include/rserver.hh]: should delete the point in deconstruct function if this point is allocated in constructor.

File: src/rutils.cc [hunk 2]
```diff
@@ -1,24 +1,78 @@
...
-    FILE *out = stdout;
-    if (lvl >= RServerLogLevel::ERRORS) {
-        out = stderr;
+RServerLog::RServerLog(RServerWorkingMode mode, std::string filename) {
+    this->mode = mode;
+    this->logLevel = RServerLogLevel::LOGS;
+
+    switch (mode) {
+        case RServerWorkingMode::STANDALONG:
+            this->printer = new StdLogPrinter();
```
Comment (wengyanqing) [inline src/rutils.cc]: delete point in deconstruct function.

File: include/rserver.hh (hunk 1 above)
Comment (HaozhouWang) [reply include/rserver.hh]: Thanks fixed.

File: src/rutils.cc (hunk 2 above)
Comment (HaozhouWang) [reply src/rutils.cc]: Thanks fixed.

**Pull request merged**